python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -521,6 +523,7 @@ def get_last_partition_key(
     def end_time_for_partition_key(self, partition_key: str) -> datetime:
         return self.time_window_for_partition_key(partition_key).end
 
+    @functools.lru_cache(maxsize=100)
```

Comment:
any methodology on this choice of maxsize? if it's turned down to 5, what happens?

Reply:
For the specific case we've been looking at recently 5 would likely do it - I guess I was trying to imagine the likely # of different partitions definitions that might be present in a real large asset graph and it seemed >5, and I saw another lru_cache(100) from https://github.com/dagster-io/dagster/pull/12981/files

But quite unscientific overall

Reply:
no methodology there, to be honest. I think 100 is fine